### PR TITLE
feat: fix security issues around customer address endpoints

### PR DESCRIPTION
**Pull request by dhargitai**
## Related Issue
VBC-426

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by scernisevs — APPROVED** [review]
LGTM! :) thank you very much

File: packages/api-client/__tests__/api/address/create.spec.ts [hunk 1]
```diff
@@ -3,25 +3,46 @@ import { contextMock } from '../../../__mocks__/context.mock';
...
+    const token = 'token123';
+    const decodedToken = { customer: { id: customerId } };
+    jwtVerifyMock.mockReturnValue(decodedToken);
+    contextMock.req = {
+      cookies: {
+        [COOKIE_KEY_CUSTOMER_DATA]: token
+      }
+    };
+    const expectedResponse = [mockedAddress];
+    contextMock.client.v3.post = (
```
Comment (WojtekTheWebDev) [inline packages/api-client/__tests__/api/address/create.spec.ts:30]: This has been implemented in a strange way and it's kind of hard to know what's happening here at the first place.
Could we change it like 
```
contextMock.client.v3.post = jest.fn(() => expectedResponse);
// ...
expect(contextMock.client.v3.post).toHaveBeenCalledWith('expectedURL', expectedParams)
```
We're using similar approach in other unit tests and I think it's easier and more descriptive :)

Comment (dhargitai) [reply packages/api-client/__tests__/api/address/create.spec.ts:30]: Sure, I agree. I already refactored this approach in other parts in this PR but it looks like not everywhere.
Thank you.

File: packages/api-client/src/helpers/auth/index.ts [hunk 2]
```diff
@@ -2,8 +2,7 @@ import { BigcommerceIntegrationContext, COOKIE_KEY_CUSTOMER_DATA } from '../..';
 const jwt = require('jsonwebtoken');
 
 export const getCustomerIdParameter = (
-  context: BigcommerceIntegrationContext,
-  params: Record<string, any>
+  context: BigcommerceIntegrationContext
```
Comment (WojtekTheWebDev) [inline packages/api-client/src/helpers/auth/index.ts:5]: Could you update also `packages/api-client/src/api/wishlist/create/index.ts` and `packages/api-client/src/api/wishlist/getAll/index.ts` files where this function is being used?

File: packages/api-client/__mocks__/context.mock.ts [hunk 3]
```diff
@@ -26,7 +26,8 @@ export const contextMock = {
     currency: '',
     lang: '',
     mediaHost: '',
-    jwtTokenExpirationDays: 2
+    jwtTokenExpirationDays: 2,
+    devtoolsAppSecret: 'secret123'
```
Comment (WojtekTheWebDev) [inline packages/api-client/__mocks__/context.mock.ts:30]: It was hardcoded in `packages/api-client/__tests__/api/wishlist/getAll.spec.ts` and `packages/api-client/__tests__/api/wishlist/create.spec.ts` files, can you also refactor them?

**Review by WojtekTheWebDev — COMMENTED** [review]
2 small remarks in comments below, then it's ready to merge 👍

**Review by WojtekTheWebDev — APPROVED** [review]
LGTM

**Pull request merged**